backend/bitboxbase/updater/updater.go

```diff
@@ -83,7 +108,150 @@ func (updater *Updater) Connect(address string, bitboxBaseID string) error {
 		return errors.New("updater: Unexpected Response Body Bytes")
 	}
 	updater.running = true
-	go listenWebsocket(updater, bitboxBaseID)
+	updater.log.Printf("connecting to base websocket")
+	client, _, err := websocket.DefaultDialer.Dial("ws://"+updater.address+"/ws", nil)
+	if err != nil {
+		updater.log.Printf("Websocket dial failed: %s", err.Error())
+	}
+	err = updater.initializeNoise(client, bitboxBaseID)
+	if err != nil {
+		return err
+	}
+
+	go listenWebsocket(client, updater, bitboxBaseID)
+	return nil
+}
+
+// initializeNoise sets up a new noise connection. First a fresh keypair is generated if none is locally found.
+// Afterwards a XX handshake is performed. This is a three part handshake required to authenticate both parties.
+// The resulting pairing code is then displayed to the user to check if it matches what is displayed on the other party's device.
+func (updater *Updater) initializeNoise(client *websocket.Conn, bitboxBaseID string) error {
+	cipherSuite := noise.NewCipherSuite(noise.DH25519, noise.CipherChaChaPoly, noise.HashSHA256)
+	keypair := updater.configGetAppNoiseStaticKeypair()
+	if keypair == nil {
+		updater.log.Info("noise static keypair created")
+		kp, err := cipherSuite.GenerateKeypair(rand.Reader)
+		if err != nil {
+			return errors.New("unable to generate a new noise keypair for the wallet app communication with the BitBox Base")
+		}
+		keypair = &kp
+		if err := updater.configSetAppNoiseStaticKeypair(keypair); err != nil {
+			updater.log.WithError(err).Error("could not store app noise static keypair")
+			// Not a critical error, ignore.
+		}
+	}
+	handshake, err := noise.NewHandshakeState(noise.Config{
+		CipherSuite:   cipherSuite,
+		Random:        rand.Reader,
+		Pattern:       noise.HandshakeXX,
+		StaticKeypair: *keypair,
+		Prologue:      []byte("Noise_XX_25519_ChaChaPoly_SHA256"),
+		Initiator:     true,
+	})
+	if err != nil {
+		return errors.New("failed to generate a new noise handshake state for the wallet app communication with the BitBox Base")
+	}
+	//Check the websocket connection
+	err = client.WriteMessage(1, []byte(opICanHasHandShaek))
+	if err != nil {
+		return errors.New("unable to write BitBox Base Handshake request to websocket")
```

Comment:
use `errp.New()` everywhere, as that adds a stack trace.

Reply:
Would something like `error.New("unable to write BitBox Base Handshake request to websocket: %s", errp.WithStack(err))` make sense?

Reply:
unfortunately not, as the 2nd arg would be turned into a string (without stack trace) before interpolation.

Just `errp.New` instead of `errors.New` is enough and will attach a stack trace.